### ISSUE TITLE
Optional Properties was broken

### ISSUE THREAD

**Issue by tuoxiansp** [vriad/zod#34]
It was worked  in 1.0.9 , but broken in 1.2.3 .

```js
export const stackProps = z
    .object({
        direction: Direction.optional(),
        horizontalAlign: Alignment.optional(),
        verticalAlign: Alignment.optional(),
        wrap: z.boolean().optional(),
        grow: z.boolean().optional(),
        shrink: z.boolean().optional(),
    })

type PropType = z.TypeOf<typeof stackProps>
```
![image](https://user-images.githubusercontent.com/3960056/79527081-4f7a7b00-8099-11ea-86a1-1bd318708d01.png)
![image](https://user-images.githubusercontent.com/3960056/79527175-818bdd00-8099-11ea-9ab1-8059f9a6b830.png)

**Comment by vriad** [comment]
Hm, can't replicate this. I assume Direction and Alignment are enums? Can you share that code with me as well? Also what version of TS are you using?

**Comment by vriad** [comment]
tag: @tuoxiansp

**Comment by tuoxiansp** [comment]
> Hm, can't replicate this. I assume Direction and Alignment are enums? Can you share that code with me as well? Also what version of TS are you using?

https://github.com/tuoxiansp/mould/blob/fix-type/components/List.tsx

At 'fix-type' branch, you can see type errors. It' OK in master.

I find some difference of _def between this two versions :

![image](https://user-images.githubusercontent.com/3960056/79528173-dfb9bf80-809b-11ea-9c04-d0f4c7b7673f.png)
sample code: [https://runkit.com/embed/74a5qve0qc61](https://runkit.com/embed/74a5qve0qc61)

**Comment by vriad** [comment]
Should be fixed in 1.2.4 - not sure how that got past my tests :/ 

Sorry about that!

**Comment by tuoxiansp** [comment]
Thanks for your awesome work!
